preview/MsixCore/msixmgr/InstallUI.cpp

```diff
@@ -340,7 +348,51 @@ HRESULT UI::ParseInfoFromPackage()
+            if (hrGetMsixPackageInfo == static_cast<HRESULT>(MSIX::Error::MissingAppxSignatureP7X))
+            {
+                TraceLoggingWrite(g_MsixUITraceLoggingProvider,
+                    "Error - Signature missing from package, calling api again with signature skip validation parameter",
+                    TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+                    TraceLoggingValue(hrGetMsixPackageInfo, "HR"));
+
+                RETURN_IF_FAILED(m_packageManager->GetMsixPackageInfo(m_path, m_packageInfo, MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_SKIPSIGNATURE));
+                m_displayErrorString = L"Ask the app developer for a new app package. This one isn't signed with a trusted certificate (0x8bad0031)";
+                SetDisplayInfo();
+                return static_cast<HRESULT>(MSIX::Error::MissingAppxSignatureP7X);
+            }
+            else if (hrGetMsixPackageInfo == static_cast<HRESULT>(MSIX::Error::CertNotTrusted))
+            {
+                TraceLoggingWrite(g_MsixUITraceLoggingProvider,
+                    "Error - Certificate is not trusted, calling api again with signature skip validation parameter",
+                    TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+                    TraceLoggingValue(hrGetMsixPackageInfo, "HR"));
+
+                RETURN_IF_FAILED(m_packageManager->GetMsixPackageInfo(m_path, m_packageInfo, MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_SKIPSIGNATURE));
+                m_displayErrorString = L"Either you need a new certificate installed for this app package, or you need a new app package with trusted certificates. Your system administrator or the app developer can help. A certificate chain processed, but terminated in a root certificate which isn't trusted (0x8bad0042)";
+                SetDisplayInfo();
+                return static_cast<HRESULT>(MSIX::Error::CertNotTrusted);
+            }
+            else if (hrGetMsixPackageInfo == static_cast<HRESULT>(MSIX::Error::FileOpen))
+            {
+                /*TraceLoggingWrite(g_MsixUITraceLoggingProvider,
+                    "Error - Invalid or corrupt File error, calling api again with signature skip validation parameter",
+                    TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+                    TraceLoggingValue(hrGetMsixPackageInfo, "HR"));
+
+                RETURN_IF_FAILED(m_packageManager->GetMsixPackageInfo(m_path, m_packageInfo, MSIX_VALIDATION_OPTION::MSIX_VALIDATION_OPTION_SKIPSIGNATURE));
+                m_displayErrorString = L"The package is either corrupted or invalid.";
+                SetDisplayInfo();
+                return static_cast<HRESULT>(MSIX::Error::FileOpen);*/
+            }
+            else
+            {
+                if (FAILED(hrGetMsixPackageInfo))
+                {
+                    SetDisplayInfo();

```

Comment:
>SetDisplayInfo [](start = 20, length = 14)

this requires m_packageInfo to be valid; if this failed, then it's not valid and this may crash

Reply:
Checked.